src/database/hooks/useLocalDatabaseHook.js

```diff
@@ -36,7 +37,17 @@ const useLocalDatabaseHook = (withMigration = false) => {
     }));
   };
 
-  const {channelInfo, channelList, channelMember, chat, user} = databaseListener;
+  const debouncedRefresh = React.useCallback(
+    _.debounce((key) => refresh(key), 50),
+    []
+  );
+
+  const refreshWithId = (key, id) => {
+    const keyWithId = `${key}_${id}`;
+    debouncedRefresh(keyWithId);
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `debouncedRefresh` function is created using `_.debounce`, which is a good practice for reducing unnecessary refreshes. However, the dependency array for `useCallback` is empty, which means the debounced function will be recreated on every render. This defeats the purpose of using `_.debounce`. To fix this, you should move the creation of the debounced function outside of the component or use `useMemo` instead of `useCallback`.
```diff
-  const debouncedRefresh = React.useCallback(
-    _.debounce((key) => refresh(key), 50),
-    []
-  );
+  const debouncedRefresh = React.useMemo(() => 
+    _.debounce((key) => refresh(key), 50),
+    [refresh]
+  );
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->